File: Data Structures & Algorithms/longest-consecutive-sequence/submission-7.java
class Solution {
    public int longestConsecutive(int[] nums) {
        int res = 0;
        Set<Integer> numSet = new HashSet<>();
        for(int num : nums){
            numSet.add(num);
        }
        for(int num : nums){
            int longest = 1;
            int current = num;
            while(numSet.contains(current + 1)){
                longest++;
                current++;
            }
            res = Math.max(res, longest);
        }
        return res;
    }
}

File: Data Structures & Algorithms/max-water-container/submission-7.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        max_water = l = 0
        r = len(heights) - 1
        while l < r:
            current = (r - l) * min(heights[l], heights[r])
            max_water = max(current, max_water)
            if heights[r] < heights[l]:
                r -= 1
            else:
                l += 1
        return max_water
        